catalyst-gateway/bin/src/metrics/endpoint.rs

```diff
@@ -0,0 +1,64 @@
+//! Metrics related to endpoint analytics.
+
+use std::sync::LazyLock;
+
+use prometheus::{register_histogram_vec, register_int_counter_vec, HistogramVec, IntCounterVec};
+
+/// Labels for the metrics
+const METRIC_LABELS: [&str; 3] = ["endpoint", "method", "status_code"];
+/// Labels for the client metrics
+const CLIENT_METRIC_LABELS: [&str; 2] = ["client", "status_code"];
+
+// Prometheus Metrics maintained by the service
+
+/// HTTP Request duration histogram.
+pub(crate) static HTTP_REQ_DURATION_MS: LazyLock<HistogramVec> = LazyLock::new(|| {
+    #[allow(clippy::ignored_unit_patterns)]
+    register_histogram_vec!(
+        "http_request_duration_ms",
+        "Duration of HTTP requests in milliseconds",
+        &METRIC_LABELS
+    )
+    .unwrap()
+});
+
+/// HTTP Request CPU Time histogram.
+pub(crate) static HTTP_REQ_CPU_TIME_MS: LazyLock<HistogramVec> = LazyLock::new(|| {
+    #[allow(clippy::ignored_unit_patterns)]
+    register_histogram_vec!(
+        "http_request_cpu_time_ms",
+        "CPU Time of HTTP requests in milliseconds",
+        &METRIC_LABELS
+    )
+    .unwrap()
+});
+
+// No Tacho implemented to enable this.
+// static ref HTTP_REQUEST_RATE: GaugeVec = register_gauge_vec!(
```

Comment:
do we need this comment ?

Reply:
For now yes, its a metric we would like, but are missing enablement code.
So I would like to leave the comment as a guide to why the metric isn't implemented.
Also, as a rule, I don't think one can have too many comments, unless they are wrong.